### PR TITLE
ENG-7695, ENG-7694: Automatic snowflake fail open and mongodb fail open

### PR DESCRIPTION
**Pull request by ptcar2009**
## Description of the change

Adds automatic fail open for snowflake and adds it to the main CFT

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Jira issue referenced in commit message and/or PR title

### Testing
Deployed it and tested e2e.

### CODE REVIEW

File: templates/cft_sidecar_failopen.yaml [hunk 1]
```diff
@@ -114,6 +125,71 @@ Parameters:
...
+    Description: Amazon EC2 instance type for the EC2 instances
+    Type: String
+  SnowflakeNGINXProxyInstanceSecurityGroup:
+    Description: "List of Security Group IDs"
+    Type: List<AWS::EC2::SecurityGroup::Id>
+  SnowflakeNGINXProxyLoadBalancerScheme:
+    Description: "EC2 network load balancer scheme ('internal' or 'internet-facing')"
+    Type: String
+    AllowedValues: ['internal', 'internet-facing']
+    Default: 'internet-facing'
```
Comment (wcmjunior) [inline templates/cft_sidecar_failopen.yaml:174]:
```suggestion
    Default: 'internal'
```

File: templates/cft_sidecar_failopen.yaml [hunk 2]
```diff
@@ -36,15 +36,26 @@ Metadata:
...
+          - SnowflakeNGINXProxyAsgDesired
+          - SnowflakeNGINXProxySubnets
+          - SnowflakeNGINXProxySSHKeyName
+          - SnowflakeNGINXProxyInstanceType
+          - SnowflakeNGINXProxyInstanceSecurityGroup
+          - SnowflakeNGINXProxyLoadBalancerScheme
+          - SnowflakeNGINXProxyLoadBalancerCertificateArn
+          - SnowflakeNGINXProxyAmiId
+          - SidecarDomainName
+          - SidecarNamePrefix
```
Comment (wcmjunior) [inline templates/cft_sidecar_failopen.yaml:58]: Please remove. Use the stack name instead.

File: templates/cft_sidecar_failopen.yaml [hunk 3]
```diff
@@ -335,3 +414,133 @@ Resources:
...
+        - Key: Name
+          Value: !Join ['-', [!Ref SidecarNamePrefix, snowflake, failopen, ASG]]
+          PropagateAtLaunch: 'true'
+        - Key: "Stack"
+          Value: !Ref "AWS::StackName"
+          PropagateAtLaunch: 'true'
+      VPCZoneIdentifier:
+        Ref: SnowflakeNGINXProxySubnets
+
+  ECSNLB:
```
Comment (wcmjunior) [inline templates/cft_sidecar_failopen.yaml:519]:
```suggestion
  NLB:
```

File: templates/cft_sidecar_failopen.yaml [hunk 4]
```diff
@@ -335,3 +414,133 @@ Resources:
...
+    Condition: IsSnowflakeRepository
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Name: !Join ['-', [!Ref AWS::StackName, lb]]
+      Type: network
+      Scheme: !Ref SnowflakeNGINXProxyLoadBalancerScheme
+      Subnets: !Ref SnowflakeNGINXProxySubnets
+      #LoadBalancerAttributes:
+      #  - Key: load_balancing.cross_zone.enabled
+      #    Value: !Ref EnableCrossZoneLoadBalancing
```
Comment (wcmjunior) [inline templates/cft_sidecar_failopen.yaml:529]: I believe we should expose the cross zone load balancing configuration to end users as parameters.

**Review by salgattcy — APPROVED** [review]

File: templates/cft_sidecar_failopen.yaml [hunk 5]
```diff
@@ -335,3 +408,200 @@ Resources:
...
+    Properties:
+      SslPolicy: ELBSecurityPolicy-2016-08
+      DefaultActions:
+        - Type: forward
+          TargetGroupArn: !Ref TargetGroup443
+      Certificates:
+        - CertificateArn: !Ref SnowflakeNGINXProxyLoadBalancerCertificate
+      LoadBalancerArn: !Ref NLB
+      Port: 443
+      Protocol: TLS
```
Comment (wcmjunior) [inline templates/cft_sidecar_failopen.yaml:607]: @ptcar2009 , please add a section in docs that states that in order for the Snowflake fail open to work, the Snowflake repo must be deployed to port `443` in the sidecar. This is not a current hard requirement and users could deploy to any port they want.

Comment (ptcar2009) [reply templates/cft_sidecar_failopen.yaml:607]: We can change this to be in the same port as the sidecar...

Comment (wcmjunior) [reply templates/cft_sidecar_failopen.yaml:607]: > We can change this to be in the same port as the sidecar...
Better yet. Could you please make this change?

**Review by wcmjunior — APPROVED** [review]

**Pull request merged**